test/scales/colorScaleTests.ts

```diff
@@ -0,0 +1,177 @@
+        scale = new Plottable.Scales.Color();
+      });
+
+      it("uses the default colors by default", () => {
+        assert.deepEqual(scale.range(), defaultColors, "The color scale uses the default colors by default");
+      });
+
+      it("uses altered colors if size of domain exceeds size of range", () => {
+        let colorRange = ["#5279c7", "#fd373e"];
+        scale.range(colorRange);
+        scale.domain(["a", "b", "c"]);
+        assert.strictEqual(scale.scale("a"), colorRange[0], "first color is used");
+        assert.strictEqual(scale.scale("b"), colorRange[1], "second color is used");
+
+        let alteredColor1 = scale.scale("c");
+        assert.notStrictEqual(alteredColor1, colorRange[0], "first color has not been reused");
+        assert.notStrictEqual(alteredColor1, colorRange[1], "second color has not been reused");
+        assert.notStrictEqual(alteredColor1, "#000000", "the color does not fallback to black when running out of colors");
+        assert.notStrictEqual(alteredColor1, "#ffffff", "the color does not fallback to white when running out of colors");
+
+        assert.operator(parseInt(alteredColor1.substr(1, 2), 16), ">", parseInt(colorRange[0].substr(1, 2), 16),
+          "The resulting color should be lighter in the red component");
+        assert.operator(parseInt(alteredColor1.substr(1, 3), 16), ">", parseInt(colorRange[0].substr(3, 2), 16),
+          "The resulting color should be lighter in the red component");
+        assert.operator(parseInt(alteredColor1.substr(1, 5), 16), ">", parseInt(colorRange[0].substr(5, 2), 16),
+          "The resulting color should be lighter in the red component");
+      });
+
+      it("interprets named color values correctly", () => {
+        scale.range(["red", "blue"]);
+        scale.domain(["a", "b"]);
+        assert.strictEqual(scale.scale("a"), "#ff0000", "red color as string should be correctly identified");
+        assert.strictEqual(scale.scale("b"), "#0000ff", "blue color as string should be correctly identified");
+        assert.deepEqual(scale.range(), ["red", "blue"], "the range itself does not convert to hex values");
+      });
+
+      it("accepts Category domain", () => {
+        scale.domain(["yes", "no", "maybe"]);
+        assert.strictEqual(scale.scale("yes"), defaultColors[0], "first color used for first option");
+        assert.strictEqual(scale.scale("no"), defaultColors[1], "second color used for second option");
+        assert.strictEqual(scale.scale("maybe"), defaultColors[2], "third color used for third option");
+      });
+    });
+
+    describe("CSS integration", () => {
+      let defaultColors = [
+        "#5279c7", "#fd373e", "#63c261", "#fad419", "#2c2b6f",
+        "#ff7939", "#db2e65", "#99ce50", "#962565", "#06cccc"
+      ];
+
+      it("accepts CSS specified colors", () => {
+        let style = d3.select("body").append("style");
+        style.html(".plottable-colors-0 {background-color: #ff0000 !important; }");
```

Comment:
I usually see hackhacks as known not-optimal workarounds that have a possible improvement through an issue fix or similar.  In this regard, I don't think this is a hackhack, but this does warrant a comment possibly.

If there is a more optimal solution, let's quickly articulate it, file and issue and then reference it through the hackhack.